src/library/crate/cratestorage.cpp

```diff
@@ -0,0 +1,665 @@
+#include "library/crate/cratestorage.h"
+
+#include "library/crate/crateschema.h"
+#include "library/dao/trackschema.h"
+
+#include "util/db/dbconnection.h"
+#include "util/db/sqltransaction.h"
+#include "util/db/fwdsqlquery.h"
+
+#include <QtDebug>
+
+
+namespace {
+
+const QString CRATETABLE_NAME = "name";
+const QString CRATETABLE_LOCKED = "locked";
+
+const QString CRATE_SUMMARY_VIEW = "crate_summary";
+
+const QString CRATESUMMARY_TRACK_COUNT = "track_count";
+const QString CRATESUMMARY_TRACK_DURATION = "track_duration";
+
+const QString kCrateTracksJoin = QString(
+        "LEFT JOIN %3 ON %3.%4=%1.%2").arg(
```

Comment:
> Support for constexpr has been added to VS 2015:

> https://msdn.microsoft.com/en-us/library/hh567368.aspx

> We had to avoid constexpr, because it wasn't supported in VS 2013. Will VS 
> 2015 definitely be used for compiling Mixxx 2.1?

Maybe @rryan or @Pegasus-RPG can tell us. 

Would QString() work in a constexpr()?